### PR TITLE
AMD Ryzen, wrong CPU Core Enumeration

### PR DESCRIPTION
**Pull request by sebastian-dev**
CPU cores not enumerated correctly.
3 threads per Core is an error -> 2 is correct
9 cores per node is an error, 8 is correct

This still needs to be tested with Ryzen 3000 and 5000.

AMD10 - cleanup, removed wrong 0x17H (Ryzen) code.

amd_node_enumeration01
amd_node_enumeration02

### CODE REVIEW

File: LibreHardwareMonitorLib/Hardware/Cpu/Amd10Cpu.cs [hunk 1]
```diff
@@ -125,11 +125,6 @@ public Amd10Cpu(int processorIndex, CpuId[][] cpuId, ISettings settings) : base(
                     }
                     break;
                 }
-                case 0x17:
-                {
-                    miscellaneousControlDeviceId = FAMILY_17H_MODEL_00_MISC_CONTROL_DEVICE_ID;
-                    break;
-                }
```
Comment (PhyxionNL) [inline LibreHardwareMonitorLib/Hardware/Cpu/Amd10Cpu.cs:132]: I don't think this file should be modified?

**Review by PhyxionNL — CHANGES_REQUESTED** [review]

File: LibreHardwareMonitorLib/Hardware/Cpu/Amd10Cpu.cs (hunk 1 above)
Comment (sebastian-dev) [reply LibreHardwareMonitorLib/Hardware/Cpu/Amd10Cpu.cs:132]: This make no sense in the AMD10 class.

Comment (PhyxionNL) [reply LibreHardwareMonitorLib/Hardware/Cpu/Amd10Cpu.cs:132]: Ah, right, I thought it was the Ryzen class.

**Comment by PhyxionNL** [comment]
I've tested it on my 3600 and it's working, but it did before. I think someone with more CCDs needs to give this a try.

**Comment by sebastian-dev** [comment]
.. my last pull is also included here...

**Comment by PhyxionNL** [comment]
> 
> 
> .. my last pull is also included here...

Please separate these two (separate PRs). I also think the HardwareAdded/Removed are not great right now as you've only added these events to network.

**Comment by sebastian-dev** [comment]
For the storage we need this feature also. Currently LHM cant add hot plugged SSD and cant remove detached Drives.

**Comment by PhyxionNL** [comment]
Adding it is fine (in a separate PR), but I think it would be more useful if only the groups that actually do this add these events. Such as a new interface that inherits from IGroup.

**Pull request merged**